packages/core/test/financial-templates/perpetual-multiparty/PositionManager.js

```diff
@@ -0,0 +1,1459 @@
+      assert.equal((await positionManager.totalPositionCollateral()).toString(), "17");
+      assert.equal((await positionManager.rawTotalPositionCollateral()).toString(), "18");
+
+      // Expected number of synthetic tokens are burned.
+      assert.equal((await tokenCurrency.balanceOf(sponsor)).toString(), "11");
+
+      // Drain excess collateral left because of precesion loss.
+      await expectAndDrainExcessCollateral();
+    });
+  });
+
+  it("Oracle swap post expiry", async function() {
+    // Approvals
+    await collateral.approve(positionManager.address, toWei("100000"), { from: sponsor });
+    await tokenCurrency.approve(positionManager.address, toWei("100000"), { from: tokenHolder });
+    await tokenCurrency.approve(positionManager.address, toWei("100000"), { from: other });
+
+    // Create one position with 200 synthetic tokens to mint with 300 tokens of collateral. For this test say the
+    // collateral is Dai with a value of 1USD and the synthetic is some fictional stock or commodity.
+    const numTokens = toWei("200");
+    const amountCollateral = toWei("300");
+    await positionManager.create({ rawValue: amountCollateral }, { rawValue: numTokens }, { from: sponsor });
+
+    // Transfer 100 the tokens from the sponsor to two separate holders. IRL this happens through the sponsor selling
+    // tokens.
+    const tokenHolderTokens = toWei("100");
+    await tokenCurrency.transfer(tokenHolder, tokenHolderTokens, {
+      from: sponsor
+    });
+    await tokenCurrency.transfer(other, tokenHolderTokens, {
+      from: sponsor
+    });
+
+    // Emergency shutdown contract to enable settlement.
+    const emergencyShutdownTime = await positionManager.getCurrentTime();
+    await financialContractsAdmin.callEmergencyShutdown(positionManager.address);
+
+    // Push a settlement price into the mock oracle to simulate a DVM vote. Say settlement occurs at 1.2 Stock/USD for the price
+    // feed. With 200 units of outstanding tokens this results in a token redemption value of: TRV = 200 * 1.2 = 240 USD.
+    await mockOracle.pushPrice(priceFeedIdentifier, emergencyShutdownTime, toWei("1.2"));
+
+    // Token holder should receive 120 collateral tokens for their 100 synthetic tokens.
+    let initialCollateral = await collateral.balanceOf(tokenHolder);
+    await positionManager.settleEmergencyShutdown({ from: tokenHolder });
+    let collateralPaid = (await collateral.balanceOf(tokenHolder)).sub(initialCollateral);
+    assert.equal(collateralPaid, toWei("120"));
+
+    // Create new oracle, replace it in the finder, and push a different price to it.
+    const newMockOracle = await MockOracle.new(finder.address, timer.address);
+    const mockOracleInterfaceName = web3.utils.utf8ToHex(interfaceName.Oracle);
+    await finder.changeImplementationAddress(mockOracleInterfaceName, newMockOracle.address, {
+      from: contractDeployer
+    });
+
+    // Settle expired should still work even if the new oracle has no price.
```

Comment:
```suggestion
    // Settle emergency shutdown should still work even if the new oracle has no price.
```

Reply:
Also, I'm glad we tested this! This test came in handy during the latest DVM upgrade